File: src/main/java/com/imdamilan/uuidauthenticator/velocity/sql/SQL.java
package com.imdamilan.uuidauthenticator.velocity.sql;

import com.imdamilan.uuidauthenticator.velocity.config.Config;
import com.imdamilan.uuidauthenticator.velocity.config.ConfigReader;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class SQL {
    public Connection connection = null;
    Config config = ConfigReader.getConfig();
    String address = config.address;
    int port = config.port;
    String database = config.databaseName;
    String tableName = config.tableName;
    String username = config.username;
    String password = config.password;

    public boolean isConnected() {
        return connection != null;
    }

    public void connect() throws SQLException, ClassNotFoundException {
        Class.forName("com.mysql.cj.jdbc.Driver");
        connection = DriverManager.getConnection("jdbc:mysql://" + address + ":" + port + "/" + database + "?useSSL=false", username, password);

        PreparedStatement statement = connection.prepareStatement("CREATE TABLE IF NOT EXISTS `" + tableName + "` (`username` VARCHAR(16) NOT NULL , `uuid` VARCHAR(36) NOT NULL, PRIMARY KEY (`username`)) ENGINE = InnoDB;");
        statement.executeUpdate();
        statement.close();
    }

    public void disconnect() throws SQLException {
        if (isConnected()) connection.close();
    }
}

File: src/main/java/com/imdamilan/uuidauthenticator/velocity/sql/SQL.kt
package com.imdamilan.uuidauthenticator.velocity.sql

import com.imdamilan.uuidauthenticator.velocity.config.ConfigReader
import java.sql.Connection
import java.sql.DriverManager

class SQL {

    private var config = ConfigReader.config
    private var address = config.databaseAddress
    private var port = config.databasePort
    private var tableName = config.databaseTable
    private var username = config.databaseUsername
    private var password = config.databasePassword

    var database = config.databaseName
    var connection: Connection? = null

    val isConnected: Boolean
        get() = connection != null

    fun connect() {
        Class.forName("com.mysql.cj.jdbc.Driver")
        connection =
            DriverManager.getConnection("jdbc:mysql://$address:$port/$database?useSSL=false", username, password)
        val statement =
            connection?.prepareStatement("CREATE TABLE IF NOT EXISTS $tableName " +
                    "(username VARCHAR(16) NOT NULL, uuid VARCHAR(36) NOT NULL," +
                    " premium BOOLEAN NOT NULL, pass VARCHAR(16) NOT NULL, PRIMARY KEY (username))")
        statement?.executeUpdate()
        statement?.close()
    }

    fun disconnect() {
        if (isConnected) connection!!.close()
    }
}